tools/fast/environment_devnet.go

```diff
@@ -0,0 +1,183 @@
+		Path:   "genesis.car",
+		Scheme: "https",
+	}
+
+	return uri.String()
+}
+
+// GenesisMiner returns a ErrNoGenesisMiner for this environment
+func (e *EnvironmentDevnet) GenesisMiner() (*GenesisMiner, error) {
+	return nil, ErrNoGenesisMiner
+}
+
+// Log returns the logger for the environment.
+func (e *EnvironmentDevnet) Log() logging.EventLogger {
+	return e.log
+}
+
+// NewProcess builds a iptb process of the given type and options passed. The
+// process is tracked by the environment and returned.
+func (e *EnvironmentDevnet) NewProcess(ctx context.Context, processType string, options map[string]string, eo EnvironmentOpts) (*Filecoin, error) {
+	e.processesMu.Lock()
+	defer e.processesMu.Unlock()
+
+	ns := iptb.NodeSpec{
+		Type:  processType,
+		Dir:   fmt.Sprintf("%s/%d", e.location, len(e.processes)),
+		Attrs: options,
+	}
+
+	e.log.Infof("New Process type: %s, dir: %s", processType, ns.Dir)
+
+	if err := os.MkdirAll(ns.Dir, 0775); err != nil {
+		return nil, err
+	}
+
+	c, err := ns.Load()
+	if err != nil {
+		return nil, err
+	}
+
+	// We require a slightly more extended core interface
+	fc, ok := c.(IPTBCoreExt)
+	if !ok {
+		return nil, fmt.Errorf("%s does not implement the extended IPTB.Core interface IPTBCoreExt", processType)
+	}
+
+	p := NewFilecoinProcess(ctx, fc, eo)
+	e.processes = append(e.processes, p)
+	return p, nil
+}
+
+// Processes returns all processes the environment knows about.
+func (e *EnvironmentDevnet) Processes() []*Filecoin {
+	e.processesMu.Lock()
+	defer e.processesMu.Unlock()
+	return e.processes[:]
+}
+
+// Teardown stops all of the nodes and cleans up the environment.
+func (e *EnvironmentDevnet) Teardown(ctx context.Context) error {
+	e.processesMu.Lock()
+	defer e.processesMu.Unlock()
+
+	e.log.Info("Teardown environment")
+	for _, p := range e.processes {
+		if err := p.core.Stop(ctx); err != nil {
+			return err
+		}
+	}
+
+	return os.RemoveAll(e.location)
+}
+
+// TeardownProcess stops the running process and removes it from the
+// environment.
+func (e *EnvironmentDevnet) TeardownProcess(ctx context.Context, p *Filecoin) error {
+	e.processesMu.Lock()
+	defer e.processesMu.Unlock()
+
+	e.log.Infof("Teardown process: %s", p.core.String())
+	if err := p.core.Stop(ctx); err != nil {
+		return err
+	}
+
+	for i, n := range e.processes {
+		if n == p {
+			e.processes = append(e.processes[:i], e.processes[i+1:]...)
+			break
+		}
+	}
+
+	// remove the provess from the process list
+	return os.RemoveAll(p.core.Dir())
+}
+
+// GetFunds retrieves a fixed amount of tokens from an environment
+func GetFunds(ctx context.Context, env Environment, p *Filecoin) error {
+	switch devenv := env.(type) {
+	case *EnvironmentDevnet:
+		var toAddr address.Address
+		if err := p.ConfigGet(ctx, "wallet.defaultAddress", &toAddr); err != nil {
+			return err
+		}
+
+		data := url.Values{}
+		data.Set("target", toAddr.String())
+
+		uri := url.URL{
+			Host:   fmt.Sprintf("faucet.%s.kittyhawk.wtf", devenv.network),
```

Comment:
should this be 
```suggestion
			Host:   fmt.Sprintf("%s.kittyhawk.wtf:9797", devenv.network),

```

Reply:
Pretty sure I made a typo earlier when trying to access this URL, looks good my bad lol